examples/audio/sound_effects.rs

```diff
@@ -0,0 +1,207 @@
+struct PlaySoundEffect {
+    name: String,
+    settings: PlaybackSettings,
+}
+
+impl Command for PlaySoundEffect {
+    fn apply(self, world: &mut World) {
+        // Access both the world and the resource we need from it using resource_scope
+        // which temporarily removes the SoundEffects resource from the world
+        world.resource_scope(|world, mut sound_effects: Mut<SoundEffects>| {
+            sound_effects.play(self.name, world, self.settings);
+        });
+    }
+}
+
+/// An "extension trait" used to make it convenient to play sound effects via [`Commands`].
+///
+/// This technique allows us to implement methods for types that we don't own,
+/// which can be used as long as the trait is in scope.
+trait SfxCommands {
+    fn play_sound_effect_with_settings(
+        &mut self,
+        name: impl AsRef<str>,
+        settings: PlaybackSettings,
+    );
+
+    fn play_sound_effect(&mut self, name: impl AsRef<str>) {
+        // This default method implementation saves work for types implementing this trait;
+        // if not overwritten, the trait's default method will be used here, forwarding to the
+        // more customizable method
+        self.play_sound_effect_with_settings(name, PlaybackSettings::DESPAWN);
+    }
+}
+
+impl<'w, 's> SfxCommands for Commands<'w, 's> {
+    // By accepting an `AsRef<str>` here, we can be flexible about what we want to accept:
+    // &str literals are better for prototyping and data-driven sound effects,
+    // but enums are nicer for special-cased effects
+    fn play_sound_effect_with_settings(
+        &mut self,
+        name: impl AsRef<str>,
+        settings: PlaybackSettings,
+    ) {
+        let name = name.as_ref().to_string();
+        self.add(PlaySoundEffect { name, settings });
+    }
+}
+
+fn spawn_button(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+
+    commands
+        .spawn(ButtonBundle {
+            style: Style {
+                width: Val::Px(300.0),
+                height: Val::Px(100.0),
+                align_self: AlignSelf::Center,
+                justify_self: JustifySelf::Center,
+                align_items: AlignItems::Center,
+                justify_content: JustifyContent::Center,
+                ..default()
+            },
+            background_color: BLUE_600.into(),
+            border_radius: BorderRadius::all(Val::Px(10.)),
+            ..default()
+        })
+        .with_children(|child_builder| {
+            child_builder.spawn(TextBundle {
+                text: Text::from_section("Generate sound effect!", TextStyle::default()),
+                ..default()
+            });
+        });
+}
+
+fn play_sound_effect_when_button_pressed(
+    button_query: Query<&Interaction, Changed<Interaction>>,
+    mut commands: Commands,
+) {
+    for interaction in button_query.iter() {
+        if *interaction == Interaction::Pressed {
+            commands.play_sound_effect("button_press");
```

Comment:
Can we use a `const` for this to lead by example?

Reply:
Yeah, that's a good idea.

Reply:
Approved; won't block on that